detekt-rules/src/test/java/org/mozilla/fenix/detektrules/PropertyAccessBannerTest.kt

```diff
@@ -0,0 +1,82 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.detektrules
+
+import io.gitlab.arturbosch.detekt.test.lint
+import io.gitlab.arturbosch.detekt.api.YamlConfig
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.Arguments.arguments
+import org.junit.jupiter.params.provider.MethodSource
+import java.util.stream.Stream
+
+internal class PropertyAccessBannerTest {
+    @Test
+    @DisplayName("non compliant property access should warn")
+    internal fun nonCompliantCodeShouldWarn() {
+        val findings =
+            PropertyAccessBanner(YamlConfig.loadResource(this.javaClass.getResource("/config.yml"))).lint(
+                NONCOMPLIANT_ACCESS.trimIndent()
+            )
+        assertThat(findings).hasSize(1)
+        assertThat(findings[0].issue.description).isEqualTo(DESCR)
+    }
+
+    @DisplayName("compliant ")
+    @MethodSource("compliantProvider")
+    @ParameterizedTest(name = "{1} should not warn")
+    internal fun testCompliantWhen(source: String) {
```

Comment:
This is super neat.  I haven't used parameterized tests much, but I really look forward to doing so.